Mobile Views

manuelmeister opened the pull request:
<img src="https://user-images.githubusercontent.com/3001985/100542567-13241580-324b-11eb-991e-60d5e4fccb7c.png" align="right" height="350">

Tried to simplify the mobile navigation and unify views (see #666)

I have reduced the use cases of the mobile view for the mvp.
I think that the following points are the primary use on mobile:
- picasso / day view
- story
- material list
- camp admin / camp information

Things I've done:
- **Searchbar**: The search bar is now gone, as it is not intuitive to search on mobile for everything. It could be revived in a global context to search for a specific enitity
- [**ContentCard**](#content-card): I changed the ContentCard component to include a toolbar that can be easily customized.  
By default, the toolbar is not displayed on the desktop within the camp context because the location is clear and you can see the currently active item in the navigation.
- [**Back Button**](#back-button): Show back button only if it's a detailpage of the last page where you can't navigate back to by the navigation bar.
- **Dialog**: Show dialogs in fullscreen on mobile
- Added icons for mobile webapp icons
- Changed Logo in README.md  

<hr width="100%" />

###### Content Card
![192 168 0 13_3000_camps_9536373a_Camp-2-Title_period_46995d25_Period-of-Camp2(iPhone 6_7_8) (1) copy](https://user-images.githubusercontent.com/3001985/100542564-0ef7f800-324b-11eb-842a-7524b1abd894.png)

manuelmeister committed:
Optimize mobile view and logo

manuelmeister committed:
Add icons for mobile browsers

manuelmeister committed:
Added ScheduleEntry number in ActivityList.vue

manuelmeister committed:
Improved rendering of the picasso interval height

manuelmeister committed:
Show dialog fullscreen when on mobile

manuelmeister committed:
Removed obsolete definitions (v-content to v-main in vuetify 2.3)

manuelmeister committed:
Improved mobile search, but is now not included as it is probably not necessary or fully clear on mobile

manuelmeister committed:
Added missing translations and fixed wrong wording

manuelmeister committed:
Added logout button on profile page on mobile

manuelmeister committed:
Removed unused fab top nav css definition

manuelmeister committed:
Added logo to home

manuelmeister committed:
Added link to collaborators on mobile (as it is not primary use case)

manuelmeister committed:
Improved Default Navigation on mobile to better align with rest

manuelmeister committed:
Prioritized entries in a tiny usability test
- Main entries are now: "eCamp Home", "Materials", "Program", "Story" and "Admin/Info" (in the future, the unprivileged members of a camp will see an info-page instead of a admin-page)
- Used camp name as a fast identifier for the program

manuelmeister committed:
Exchanged mobile activities and camp search with period switcher and optimized the viewmode switcher

manuelmeister committed:
Linted and fixed missing definitions

carlobeltrame commented on README.md:
```diff
@@ -1,11 +1,9 @@
-  <a href="https://coveralls.io/github/ecamp/ecamp3?branch=devel"><img src="https://coveralls.io/repos/github/ecamp/ecamp3/badge.svg?branch=devel" alt="Coverage Status"></a>
-  <a href="https://github.com/ecamp/ecamp3/blob/devel/LICENSE"><img src="https://badgen.net/github/license/ecamp/ecamp3" alt="License"></a>
-</p>
+# ![Logo](frontend/public/logo.svg) eCamp v3
+[![Build Status](https://travis-ci.com/ecamp/ecamp3.svg?branch=devel)](https://travis-ci.com/ecamp/ecamp3)
+[![Coverage Status](https://coveralls.io/repos/github/ecamp/ecamp3/badge.svg?branch=devel)](https://coveralls.io/github/ecamp/ecamp3?branch=devel)
+[![License](https://badgen.net/github/license/ecamp/ecamp3)](https://github.com/ecamp/ecamp3/blob/devel/LICENSE)
```
Those were html tags for a reason... https://github.com/ecamp/ecamp3/blob/8ee1f9632660288a80b773b63a68f3455a25663b/README.md doesn't look as nice anymore ;)

carlobeltrame commented on README.md:
```diff
@@ -25,7 +23,7 @@ The following commonly requested improvements over eCamp v2 are already implemen
 eCamp v3 is made up of a backend based on the PHP framework Laminas (which offers a HAL JSON API) as well as a Vue.js frontend and some other smaller services.
 
-#### Want to help?
+### Want to help?
```
If you change this, please also change the corresponding headings in CONTRIBUTING.md

manuelmeister committed:
Reverted changes to make raw readme more readable

https://github.com/ecamp/ecamp3/pull/697#discussion_r531015710

manuelmeister committed:
Changed CONTRIBUTING to match README

manuelmeister committed:
Removed unused prop and fixed type

manuelmeister committed:
Extracted CampProgramViewModeSwitcher

manuelmeister committed:
Made switch more accesible and added generic way for empty states

manuelmeister committed:
Fix naming of css class

manuelmeister committed:
Improved meaning of camp admin

manuelmeister committed:
Optimized mobile view, Back Button depends on request & standard way to style toolbar inside ContentCard
- Show back button only if it's a detailpage of the last page where you can go back to the overview. Otherwise if we show it all the time and you navigate via navigation for instance, the mental model of the "eCamp Back Button" is corrupted, as it doesn't navigate to the overview but to the last page. If you want to go back to last page use the browser back button.
- Optimized ContentCard.vue to display a standard toolbar on mobile and on desktop only when necessary (if location is unclear)

manuelmeister committed:
Add logout to global.button

manuelmeister committed:
Fixed intervalHeight on mobile

manuelmeister committed:
Fixed overflow on mobile viewports

manuelmeister committed:
Made toolbar sticky on mobile

manuelmeister committed:
Added generic translations for no results

manuelmeister committed:
Removed search on desktop as well as mobile.
In mobile view it doesn't make sense, as it is not intuitive to search for your activities/schedule entries/camps. The search would be usefull on global ecamp context and not inside the camp context.

manuelmeister committed:
Force toolbar and fixed wrong slot

manuelmeister committed:
Fixed wrong route on mobile

manuelmeister commented on README.md:
```diff
@@ -1,11 +1,8 @@
-  <a href="https://coveralls.io/github/ecamp/ecamp3?branch=devel"><img src="https://coveralls.io/repos/github/ecamp/ecamp3/badge.svg?branch=devel" alt="Coverage Status"></a>
-  <a href="https://github.com/ecamp/ecamp3/blob/devel/LICENSE"><img src="https://badgen.net/github/license/ecamp/ecamp3" alt="License"></a>
-</p>
+<p align="center"><img src="frontend/public/logo.svg" alt="Logo" width="250"></p>
+<h1 align="center">eCamp <sup><small>v3</small></sup></h1>
+<p align="center"><a href="https://travis-ci.com/ecamp/ecamp3"><img src="https://travis-ci.com/ecamp/ecamp3.svg?branch=devel" alt="Build Status"></a> <a href="https://coveralls.io/github/ecamp/ecamp3?branch=devel"><img src="https://coveralls.io/repos/github/ecamp/ecamp3/badge.svg?branch=devel" alt="Coverage Status"></a> <a href="https://github.com/ecamp/ecamp3/blob/devel/LICENSE"><img src="https: ...
```
Changed ecamp title to be machine readable, instead of an image

manuelmeister commented on frontend/public/index.html:
```diff
@@ -1,21 +1,30 @@
+<head>
+ <meta charset="utf-8">
+ <meta http-equiv="X-UA-Compatible" content="IE=edge">
+ <meta name="apple-mobile-web-app-capable" content="yes">
+ <meta name="mobile-web-app-capable" content="yes">
+ <link rel="apple-touch-icon" sizes="180x180" href="<%= BASE_URL %>browserassets/apple-touch-icon.png">
+ <link rel="icon" type="image/png" sizes="32x32" href="<%= BASE_URL %>browserassets/favicon-32x32.png">
+ <link rel="icon" type="image/png" sizes="16x16" href="<%= BASE_URL %>browserassets/favicon-16x16.png">
+ <link rel="mask-icon" href="<%= BASE_URL %>browserassets/safari-pinned-tab.svg" color="#263238">
+ <meta name="msapplication-TileColor" content="#263238">
+ <meta name="theme-color" content="#263238">
+ <link rel="manifest" href="<%= BASE_URL %>manifest.webmanifest">
```
Added definitions to optimize webapp on mobile

manuelmeister commented on frontend/src/locales/de.json:
```diff
@@ -221,6 +222,10 @@
       "update": "Aktualisieren"
     },
     "changeLanguage": "Sprache ändern",
+    "empty": {
+      "list": "Leer | Noch keine {entities}",
+      "result": "Kein Resultat | Keine Resultate | Keine Resultate für {search} | Keine {search} gefunden"
+    },
```
Made a generic way to display empty states

carlobeltrame replied:
How can we use this four-way pluralization?

manuelmeister replied:
Hm 🤔 I think that I have misused this feature. It handles several issues in one key.

manuelmeister replied:
Would you remove that again?

carlobeltrame replied:
I think so, as long as we can't use it. As far as I understand, only languages with other pluralization rules, such as russian, can use more than 3 pluralized entries. If you really need both "Keine Resultate für {search}" and "Keine {search} gefunden", I'd propose you create two different translation keys.

manuelmeister committed:
Moved expansion-panel border definition to global.scss

manuelmeister committed:
Implicit descendant

manuelmeister committed:
It should work for sheets as well

manuelmeister committed:
Merge branch 'devel' of github.com:ecamp/ecamp3 into feature/mobile-views

manuelmeister committed:
Reorganized global empty keys, used 10 as placeholder for many (this works in russian as it is the same as the plural general quantity)

manuelmeister committed:
Merge branch 'devel' of github.com:ecamp/ecamp3 into feature/mobile-views

carlobeltrame commented on CONTRIBUTING.md:
```diff
@@ -65,8 +66,9 @@ Alternatively you can
 
-
-## Mitarbeit (deutsche Übersetzung)
+<section lang="de">
```
If you add this, consider also adding it to the english section, and to the german and english section in the README.md

carlobeltrame commented on frontend/public/index.html:
```diff
@@ -1,21 +1,30 @@
+</head>
+<body>
+<noscript>
+ <strong>Sorry, eCamp funktioniert nicht ohne JavaScript. Bitte aktiviere JavaScript um fortzufahren.</strong>
```
Maybe we should write this notice in English? Or in multiple languages?

carlobeltrame commented on frontend/public/manifest.webmanifest:
```diff
@@ -0,0 +1,31 @@
+{
+  "short_name": "eCamp",
+  "name": "eCamp v3",
+  "description": "Lagerplanung",
```
Maybe describe this in English?

manuelmeister replied:
Removed it, it isn't used by browsers

carlobeltrame commented on frontend/src/components/camp/Picasso.vue:
```diff
@@ -172,6 +172,9 @@ export default {
       return this.period().camp()
+    },
+    computedIntervalHeight () {
+      return this.intervalHeight !== 0 ? this.intervalHeight : this.$vuetify.breakpoint.xsOnly ? (window.innerHeight - 140) / 19 : (window.innerHeight - 174) / 19
```
Rethink this calculation. When resizing the window height, this does not react. On very small screen heights, the grid is way too tight.

carlobeltrame commented on frontend/src/views/camp/NavigationCamp.vue:
```diff
@@ -26,33 +26,33 @@
       </v-btn>
       <v-btn :to="campRoute(camp(), 'story')" text>
-        <v-icon :left="$vuetify.breakpoint.mdAndUp">mdi-book-open-page-variant</v-icon>
+        <v-icon :left="$vuetify.breakpoint.mdAndUp">mdi-book</v-icon>
```
If you change this, you should also change the activityContent icon as well (when adding a new story content to an activity). But to be honest, I liked the open book icon better... This one is a little abstract and if I focus it means "bookmark" to me.

manuelmeister replied:
Tried out different icons:
![localhost_3000_(iPhone 6_7_8)](https://user-images.githubusercontent.com/3001985/102702454-2aab5880-4263-11eb-81f2-e1c9bcac0e67.png)

carlobeltrame commented on frontend/src/components/layout/ContentCard.vue:
```diff
@@ -4,29 +4,52 @@ Displays the content wrapped inside a card.
+               elevation="0"
+               dense>
+      <v-toolbar-items>
+        <button-back v-if="back || $vuetify.breakpoint.xsOnly && !!$route.query.isDetail" class="ml-n4" />
```
As discussed: Remove back buttons for now, except for nested settings navigation on mobile. The top navigation on desktop and bottom navigation on mobile should suffice for the most accessed places.

carlobeltrame replied:
The back buttons are still here on at least the camp list, the picasso view and the activity detail view. Did I misunderstand the outcome of our discussion?

carlobeltrame commented on frontend/src/components/camp/PeriodSwitcher.vue:
```diff
@@ -0,0 +1,49 @@
+  <v-toolbar-items v-if="period().camp().periods().items.length > 1">
+    <v-menu offset-y>
+      <template v-slot:activator="{ on, attrs, value }">
+        <v-btn
```
Consider removing the left padding on this button so the text is properly left aligned with the contents of the ContentCard

manuelmeister committed:
Set min height of intervalHeight and made used reactive innerHeight

manuelmeister committed:
Fix path of logos

carlobeltrame commented on frontend/src/scss/global.scss:
```diff
@@ -50,3 +50,13 @@ body {font-feature-settings: "ss03", "calt", "rlig", "kern", "cv05", "cv07", "cv
 .flex-full {
 	flex: 1 1 0;
 }
+
+.v-sheet > .v-expansion-panels,
+.v-card > .v-expansion-panels {
+	border-top: 1px solid #eee;
+	border-bottom: 1px solid #eee;
+
+	> .v-expansion-panel + .v-expansion-panel {
+		border-top: 1px solid #eee;
+	}
+}
```
If you restyle this, also use the `accordion` mode on the `v-expansion-panels` so that the panels don't move down when opening.

carlobeltrame replied:
This is not fixed yet: 
![moving-line](https://user-images.githubusercontent.com/7566995/102724835-c8aa2c00-4312-11eb-9cee-b232caa0d1ca.gif)
Could you remove the additional top margin that the expansion panel has in the open state? That way it doesn't look as if the top period grows when I open the bottom period.

carlobeltrame commented on frontend/src/views/camp/Story.vue:
```diff
@@ -3,40 +3,45 @@ Admin screen of a camp: Displays details & periods of a single camp and allows t
+                    :period="period"
+                    :editing="editing" />
+    </v-expansion-panels>
+    <v-card-actions>
```
The stoy content can potentially get very long. Do you see any way to keep the print button at the top?

manuelmeister replied:
<img src="https://user-images.githubusercontent.com/3001985/102702265-47df2780-4261-11eb-8baa-9a097af733bd.png" width=200px align=right>

I think this is not the first thing you want to do on this page, I don't know where to put it instead. Also where (if at all) can I put it on mobile?

carlobeltrame replied:
Maybe in a three-dot menu next to the edit switch?

carlobeltrame reviewed (changes requested):
Very nice work! As discussed, a few changes I'd still like to see

manuelmeister committed:
Compacter view with expansion panel

manuelmeister committed:
Merge branch 'devel' of github.com:ecamp/ecamp3 into feature/mobile-views

manuelmeister committed:
Removed language section and added link

manuelmeister committed:
Added more languages in noscript

manuelmeister committed:
Is not used by browsers today, may as well postpone translation issue :)

manuelmeister committed:
Optimized story icon and labels

manuelmeister committed:
Removed the isDetail from every view, except the detail views in camp admin on mobile

manuelmeister committed:
Removed padding from period switcher

manuelmeister committed:
Hide label of add button on camp admin

carlobeltrame commented on CONTRIBUTING.md:
```diff
@@ -1,8 +1,10 @@
+<span lang="en">Thank you for wanting to help out!</span> <em lang="de">Danke dass du mithelfen möchtest!</em>
 
-*(deutsche Übersetzung weiter unten)*
+<em lang="de">Die <a link="#deutsch">deutsche Übersetzung</a> findest du weiter unten.</em>
```
```suggestion
<em lang="de">Die <a href="#deutsch">deutsche Übersetzung</a> findest du weiter unten.</em>
```

manuelmeister committed:
Update CONTRIBUTING.md

Co-authored-by: Carlo Beltrame <carlo@beltra.me>

carlobeltrame reviewed (dismissed):
Sorry about the comments, I forgot to pull. Everything looks fine now. The story print button is not really urgent, I think this page might be redesigned some time anyways.

manuelmeister committed:
Improve presentation of story view on mobile

manuelmeister committed:
Merge branch 'feature/mobile-views' of github.com:manuelmeister/ecamp3 into feature/mobile-views

manuelmeister commented:
@carlobeltrame no problem. I tried it with the 3 dots and it works and is okay, but I agree that it might need another general redesign.
